tests/foreman/virtwho/ui/test_esx.py

```diff
@@ -54,6 +57,14 @@ def form_data():
 
 
 class TestVirtwhoConfigforEsx:
+    @pytest.mark.tier2
```

Comment:
The mark should be applied to the class, not this setup function.

Secondly, is `setup_method` automatically called in pytest? This looks like a unittest pattern to me, and we're executing with pytest.

The method should be decorated as a fixture, and if you want every function to use it without having to include it, you should also decorate with `autouse=True`, or apply the `usefixtures` mark on the class.

Reply:
> The mark should be applied to the class, not this setup function.
> 
> Secondly, is `setup_method` automatically called in pytest? This looks like a unittest pattern to me, and we're executing with pytest.
> 
> The method should be decorated as a fixture, and if you want every function to use it without having to include it, you should also decorate with `autouse=True`, or apply the `usefixtures` mark on the class.

@mshriver the setup_method works well, in fact, what I want to is to run this setup_method before run the test_XXX method in order to make a clean env for every test_XX method only in this class.
If there a demo which I can refer ?